delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/CompositeViews.scala

```diff
@@ -128,6 +494,109 @@ object CompositeViews {
       case c: TagCompositeView       => tag(c)
       case c: DeprecateCompositeView => deprecate(c)
     }
+  }
+
+  /**
+    * Constructs a new [[CompositeViews]] instance.
+    */
+  def apply(
+      config: CompositeViewsConfig,
+      eventLog: EventLog[Envelope[CompositeViewEvent]],
+      permissions: Permissions,
+      orgs: Organizations,
+      projects: Projects,
+      acls: Acls
+  )(implicit
+      uuidF: UUIDF,
+      clock: Clock[UIO],
+      as: ActorSystem[Nothing],
+      baseUri: BaseUri,
+      rmr: RemoteContextResolution
+  ): Task[CompositeViews] = {
+
+    def validateAcls(cpSource: CrossProjectSource) = {
+      val aclAddress = AclAddress.Project(cpSource.project)
+      acls.fetchWithAncestors(aclAddress).map(_.exists(cpSource.identities, resources.read, aclAddress)).flatMap {
+        case true  => IO.unit
+        case false => IO.raiseError(CrossProjectSourceForbidden(cpSource))
+      }
+    }
+
+    def validateProject(cpSource: CrossProjectSource) = {
+      projects.fetch(cpSource.project).leftMap(_ => CrossProjectSourceProjectNotFound(cpSource))
```

Comment:
`.mapError`. Try to avoid using the cats alternative when the current implementation of IO sufices.